test/unit/matrix/test_panel.cpp

```diff
@@ -42,14 +45,31 @@ struct PanelTest : public TestWithCommGrids {};
 
 TYPED_TEST_SUITE(PanelTest, MatrixElementTypes);
 
+// Helper for checking if current rank, along specific Axis, locally stores just an incomplete tile,
+// i.e. a tile with a size < blocksize
+template <Coord Axis>
+bool doesThisRankOwnsJustIncomplete(const matrix::Distribution& dist) {
+  // an empty matrix does not fall in this specific edge-case
+  if (dist.nrTiles().isEmpty())
+    return false;
+
+  // look at the last tile along panel_axis dimension, and see if its size is full or not
+  const GlobalTileIndex last_tile_axis(Axis, dist.nrTiles().get(Axis) - 1);
+  const bool is_last_tile =
+      dist.rankIndex().get(Axis) == dist.template rankGlobalTile<Axis>(last_tile_axis.get(Axis));
+  return is_last_tile && dist.localNrTiles().get(Axis) == 1 &&
+         dist.tileSize(last_tile_axis).get(Axis) != dist.blockSize().get(Axis);
+}
+
 struct config_t {
   const GlobalElementSize sz;
   const TileElementSize blocksz;
-  const GlobalTileIndex offset;
+  const GlobalTileIndex offset = {0, 0};
 };
 
 std::vector<config_t> test_params{
-    {{0, 0}, {3, 3}, {0, 0}},  // empty matrix
+    {{0, 0}, {3, 3}},  // empty matrix
+    {{8, 5}, {3, 3}},
```

Comment:
This test-case, when distributed, resulted in edge cases, in particular in case of empty local matrices or just incomplete tiles available locally, for the next two tests:
- ExternalTiles
- Shrink

I opted for adding a check for this edge-cases and just skip them without over-complicating the implementation of the test.

Do you think it is ok?

Reply:
It is fine for now.